### ISSUE TITLE
Request for Guidance on Deno Template Removal

### ISSUE THREAD

**Issue by zkdiff** [remix-run/react-router-templates#5]
> **Note:** If the removal of the Deno template is transient, please disregard this issue—I understand changes like this happen during early/rapid/active development.

The Deno template for `@remix-run/react-router-templates` was recently created and then removed. I would like to understand the implications of this change for developers who may have been interested in using the template.

I'm seeking clarification on the following points, as it is unclear what the removal signals:

* Is Deno v2 an officially supported target runtime/platform for RR v7?
* Is a separate template no longer warranted due to Deno v2's Node.js compatibility improvements?
* If I'd like to get started using RR v7 on Deno, where would you recommend I start?

Please let me know if additional information would help in addressing this issue.

### Additional Context

Any insight or future roadmap details regarding Deno support would be greatly appreciated.

I'm also interested in contributing to the Deno template to help reintroduce or maintain support, if that would be helpful.

**Comment by redabacha** [comment]
yeah this would be much appreciated knowing the remix's team stance on Deno v2 support. i've made a working template for RR v7 + Deno v2 based on the templates here (it uses typescript for the app directory so the language server plugin works as expected): https://github.com/redabacha/react-router-deno-template/. however it would be nice to see it officially supported.

**Comment by diecodev** [comment]
@brookslybrand deploy failed bc of this:
- https://github.com/denoland/deno/issues/27475

**Comment by brookslybrand** [comment]
> [@brookslybrand](https://github.com/brookslybrand) deploy failed bc of this:
> 
> * [The requested module `http2` does not provide an export named `Http2ServerRequest` denoland/deno#27475](https://github.com/denoland/deno/issues/27475)

Nice!

If this gets fixed, I'm happy for someone to open up a PR to add back  the deno template

**Comment by zkdiff** [comment]
[dirty-shark-76](https://dash.deno.com/playground/dirty-shark-76) is reporting:

```
Deno/2.1.4 
 Hono! Hono! Hono! Hono! 
  Hono says this is Deno v2.x
```

So maybe it's time to revisit. I'll try to give it a spin tonight.

**Comment by diecodev** [comment]
@zkdiff As far as I know, the problem is not related to Hono but to the import of react-router in the entry server to stream component data using Node.js APIs.

However, I'll also take a look at it today :D

**Comment by redabacha** [comment]
> [@redabacha](https://github.com/redabacha) if your template is similar to the existing ones/[the one we originally had](https://github.com/remix-run/react-router-templates/tree/855176cee6355ca2ff6eb0bedc89631b7fff994a/deno-custom-server), please definitely feel free to open up a PR and I'll try to get it in

sorry i haven't checked on this thread in a while and yep it's pretty similar (except the custom express server stuff) and i've updated it so it only relies on the deno typechecker now. i will open a pr to add the deno template along with a guide for adding a custom express/hono server that should work the same in both dev and prod.

**Comment by brookslybrand** [comment]
I'm going to close this issue since the "request for guidance" has been given

If someone (@redabacha?) wants to open a PR with a working Deno example please tag me and I'll do my best to help get it in 🙂
